src/eos-google-chrome-app

```diff
@@ -151,8 +133,42 @@ class GoogleChromeFlatpakLauncher:
             if not runtime_path or not os.path.exists(runtime_path):
                 exit_with_error("Could not find Chrome's external runtime directory")
 
-        # Both application and runtime directories found if reached.
-        app_info = {'app_lib_path': app_lib_path, 'runtime_path': runtime_path}
+            runtime_path = os.path.join(runtime_path, 'files')
+            if os.path.exists(runtime_path):
+                exit_with_error("Could not find Chrome's binaries in external runtime directory")
+
+        return runtime_path
+
+    def _get_chrome_app_info(self):
+        app_lib_path = None
+        runtime_path = None
+
+        app = find_flatpak_ref(self._installation, self.FLATPAK_CHROME_APP_ID)
+        if not app:
+            logging.info("Chrome application is not installed")
+            return None
+
+        app_path = app.get_deploy_dir()
+        if not app_path or not os.path.exists(app_path):
+            exit_with_error("Could not find Chrome's application directory")
+
+        app_lib_path = os.path.join(app_path, 'files/lib/x86_64-linux-gnu')
+        if not os.path.exists(app_lib_path):
+            exit_with_error("Could not find bundled libraries for Chrome")
+
+        if app_path and not app_lib_path:
+            exit_with_error("Chrome's installation directory found, but not bundled libraries")
+
+
+        chrome_path = os.path.join(app_path, 'files', 'extra')
+        # We check for legacy runtimes to maintain backward-compatibility for users without
+        # support for native external apps (Flatpak <0.6.13)
+        if not os.path.exists(chrome_path):
+            exit_with_error("Could not find 'extra data' directory in Chrome's Flatpak; falling "
```

Comment:
You should not exit here, or you'll never return the legacy path. `logging.warning()` would probably be better